documentation/providers/loopia.md

```diff
@@ -0,0 +1,214 @@
+a domain, do so from the Loopia control panel.
+ * `addDomain`
+ * `transferDomain` (to Loopia)
+
+This effectively means that this plugin does not access registrar functions.
+
+## Errors
+
+You may occasionally see this error
+```text
+HTTP Post Error: Post "https://api.loopia.se/RPCSERV": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
+```
+
+The API endpoint didn't answer. Try again. 🤷
+
+
+## Configuration
+
+To use this provider, add an entry to `creds.json` with `TYPE` set to `LOOPIA`
+along with your Loopia API login credentials.
+
+Example:
+
+{% code title="creds.json" %}
+```json
+{
+  "loopia": {
+    "TYPE": "LOOPIA",
+    "username": "your-loopia-api-account-id@loopiaapi",
+    "password": "your-loopia-api-account-password",
+    "debug": "true" // Set to true for extra debug output. Remove or set to false to prevent extra debug output. 
+  }
+}
+```
+{% endcode %}
+
+### Variables
+
+* `username` - string - your @loopiaapi created username
+* `password` - string - your loopia API password
+* `debug` - string - Set to true for extra debug output. Remove or set to false to prevent extra debug output. 
+* `rate_limit_per` - string - See [Rate Limiting](#rate-limiting) below.
+* `region` - string - See [Regions](#regions) below.
+* `modify_name_servers` - string - See [Modify Name Servers](#modify-name-servers) below.
+* `fetch_apex_ns_entries` - string - See [Fetch NS Entries](#fetch-apex-ns-entries) below.
+
+There is no test endpoint. Fly free, grasshopper.
+
+Turning on debug will show the XML requests and responses, and include the
+username and password from your `creds.json` file. If you want to share these, 
+like for a Github issue, be sure to redact those from the XML.
+
+### Fetch Apex NS Entries
+
+... or use locally hard-coded variables:
+
+```go
+  defaultNS1       = "ns1.loopia.se."
+  defaultNS2       = "ns2.loopia.se."
+```
+
+API calls to loopia can be expensive time-wise. Set this to "false" (off) to
+skip the API call to fetch the apex (`@`) entries, and use Loopia's default NS
+servers.
+
+This setting defaults to "true" (on).
+
+### Modify Name Servers
+
+Enabling this setting to "true" (on) allows you to modify NS entries.
+
+Loopia is weird. NS entries are inaccessible in the control panel. But you can see them.
+Perhaps dnscontrol added an NS that you cannot delete now? Toggle this setting to
+"true" in order to treat all NS entries as any other - making them accessible
+to modification. Beware the consequences of changing from default NS entries. Likely
+nothing will happen since the glue records provided won't match those in the domain,
+and you will need to manually inform Loopia of this so they can update the glue records.
+
+In short: enable this setting to be able to delete NS entries. No `NS()` in your
+`dnsconfig.js`? Existing ones will be deleted. Have some `NS()` or `NAMESERVER()`
+entries? They'll be added.
+
+This setting defaults to "false" (off).
+
```

Comment:
NOTE: This setting is global.

(do you want to implement this on a per-domain basis?  I can show you how to move that setting from creds.json to be a meta-parameter for the domain)

Reply:
AI think that would be nice. It shouldn't take me too long to implement. Maybe I could add it later. Are there any examples already in code or do you have boiler-plate?